services/git-service-impl/src/main/java/io/fabric8/launcher/service/gitlab/impl/GitLabServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+package io.fabric8.launcher.service.gitlab.impl;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.service.git.api.GitHook;
+import io.fabric8.launcher.service.git.api.GitRepository;
+import io.fabric8.launcher.service.git.api.GitUser;
+import io.fabric8.launcher.service.git.api.ImmutableGitHook;
+import io.fabric8.launcher.service.git.api.ImmutableGitRepository;
+import io.fabric8.launcher.service.git.api.ImmutableGitUser;
+import io.fabric8.launcher.service.git.api.NoSuchHookException;
+import io.fabric8.launcher.service.git.api.NoSuchRepositoryException;
+import io.fabric8.launcher.service.git.impl.AbstractGitService;
+import io.fabric8.launcher.service.gitlab.api.GitLabService;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+class GitLabServiceImpl extends AbstractGitService implements GitLabService {
+
+    private static final MediaType APPLICATION_FORM_URLENCODED = MediaType.parse("application/x-www-form-urlencoded");
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    GitLabServiceImpl(TokenIdentity identity) {
+        super(identity);
+    }
+
+    @Override
+    public GitRepository createRepository(String repositoryName, String description) throws IllegalArgumentException {
+        StringBuilder content = new StringBuilder();
+        content.append("name=").append(repositoryName);
+        if (description != null && !description.isEmpty()) {
+            content.append("&description=").append(description);
+        }
+        Request request = request()
+                .post(RequestBody.create(APPLICATION_FORM_URLENCODED, content.toString()))
+                .url("https://gitlab.com/api/v4/projects")
```

Comment:
Can we make URL configurable? This way we would support on-premise gitlab installation.